File: openresty/cdn-system/api/models/domain.go
package models

import "time"

type Domain struct {
	ID        int64     `json:"id" gorm:"primaryKey"`
	UserID    int64     `json:"user_id" gorm:"uniqueIndex:idx_user_domain"`
	Name      string    `json:"name" gorm:"uniqueIndex:idx_user_domain"`
	Cname     string    `json:"cname"`
	Status    int       `json:"status"` // 1: Active, 0: Disabled
	
	// Legacy storage (if exists in DB)
	OriginsRaw string         `json:"-" gorm:"column:origins"`
	Origins    []DomainOrigin `json:"origins" gorm:"foreignKey:DomainID"`
	
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

File: cdn-system/cnn.net/src/Cnn.Common/Domain/Entities/Domain.cs
using SqlSugar;

namespace Cnn.Domain.Entities;

[SugarTable("domains")]
public class Domain
{
    [SugarColumn(IsPrimaryKey = true, IsIdentity = true)]
    public long Id { get; set; }

    public long? UserId { get; set; }

    public string? Name { get; set; }

    public string? Cname { get; set; }

    public int? Status { get; set; }

    [SugarColumn(ColumnName = "origins")]
    public string? OriginsRaw { get; set; }

    [SugarColumn(ColumnName = "created_at")]
    public DateTime? CreatedAt { get; set; }

    [SugarColumn(ColumnName = "updated_at")]
    public DateTime? UpdatedAt { get; set; }

    [SugarColumn(IsIgnore = true)]
    public List<DomainOrigin>? Origins { get; set; }
}
